### DISCUSSION TITLE
need help with slug random id

### DISCUSSION THREAD

**Discussion by vishalkadam47** [enchant97/note-mark#179] [General]
Hi, I like this random slug id option, this request is not related to note-mark, as I'm not a developer but still I'm trying to build a nextjs app for now if a user enters a project name it creates slug with same name. how do I add this random id in my code can i get some help. if its ok for anyone i can add you in my repo. PR would be helpful. 

[Screencast from 13-01-24 07:44:47 PM IST.webm](https://github.com/enchant97/note-mark/assets/107353260/07a18b19-3fb4-44d7-82bb-2936ea07e171)

**Comment by enchant97** [comment]
Hi, I would be happy to help. This is the code that generates the slugs: https://github.com/enchant97/note-mark/blob/03dd3dbe578e5edc859e1320a4f7d89cfd25e9c1/frontend/src/core/helpers.ts#L1-L14

**Comment by vishalkadam47** [comment]
thanks. solved the issue somehow 
Not exactly sure what i did but its working. 

```
import { randomBytes } from 'crypto';
import slugify from 'slugify';

export function toSlugWithSuffix(name: string) {

  const random = randomBytes(5).toString('hex');

  const slug = slugify(name);

  return `${slug}-${random}`;

}
```


```
import { toSlugWithSuffix } from '@/lib/slugUtils';

// Generate slug on name change
    const generateSlug = (name: string) => {
      return toSlugWithSuffix(name);
    }

    useDebounce(() => {
      const slug = toSlugWithSuffix(data.name);

      setData(prev => ({
        ...prev,
        slug
      }));
    }, 200, [data.name]);
```
